lib/shopping_cart.rb

```diff
@@ -0,0 +1,53 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize(name, capacity)
+    @name,  = name,
+    @capacity = capacity.strip.tr('^0-9', '').to_i
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    cart_details = {
```

Comment:
no need to save this into a variable since it's the last line of your method anyway.